### PR TITLE
Employee Roles Dropdown

### LINKED ISSUE #35: Add roles dropdown
- [x] fetch employee roles
- [x] add custom dropdown

### PR DESCRIPTION
**Pull request by alisondraV**
## Description

This PR brings in the following changes:

- custom dropdown
- add `mutate` to `employeeService` for data revalidation
- further Employees page styling
- Employees page translation

## Issue

Fixes #35 & #42

### CODE REVIEW

**Review by ANDREYDEN — CHANGES_REQUESTED** [review]
Nice work, here are some comments:
- don't bother if it's too much work, but it would be nice if the modal would not exceed the height of the screen and the fields list would be scrollable (instead of the whole thing)
image
- Should the employee table show the role name vs role_id? Or is that noted in a separate issue?

**Comment by alisondraV** [comment]
> Nice work, here are some comments:
> 
> * don't bother if it's too much work, but it would be nice if the modal would not exceed the height of the screen and the fields list would be scrollable (instead of the whole thing)
>   image
> * Should the employee table show the role name vs role_id? Or is that noted in a separate issue?

Yes, that's something that involves more work regarding the Table, so I wanted to do it in the other PR.

File: interfaces/Dropdown.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    label: string
+}
+
+export interface IDropdown {
+    label: string
+    name: string
+    data: IDropdownOption[]
+    defaultOption: string
+    dropdownClass?: string
+}
```
[the remark this answers is not in the document]
Comment (alisondraV) [reply interfaces/Dropdown.ts:12]: Sure, let's discuss

File: components/employees/index/AddEmployeeModal.tsx [hunk 2]
```diff
@@ -38,23 +41,36 @@ const AddEmployeeModal: React.FC<IAddEmployeeModal> = ({ addEmployee, toggleModa
...
-            role_id: 1,
+            role_id: data.role_id,
             birth_date: data.birth_date === '' ? null : data.birth_date,
             salary: data.salary,
             income_percentage: data.income_percentage
         }
         await addEmployee(newEmployee)
         toast('🦄 An employee has been successfully added')
+
+        revalidateEmployees(newEmployee)
```
[the remark this answers is not in the document]
Comment (alisondraV) [reply components/employees/index/AddEmployeeModal.tsx:65]: I'm not sure, where to put the second call

**Review by ANDREYDEN — APPROVED** [review]

**Pull request merged**